seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -540,49 +716,56 @@ def test_create_project_from_workspace(self, mock_tempfile, mock_mv_file, mock_s
+            mock_load_file, mock_airtable_logger, mock_utils_logger):
+
         url = reverse(create_project_from_workspace, args=[TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME])
+        url2 = reverse(create_project_from_workspace, args=[TEST_WORKSPACE_NAMESPACE, TEST_NO_PROJECT_WORKSPACE_NAME2])
+
+        self._test_loading_delay_email(url, url2, REQUEST_BODY, REQUEST_BODY, mock_send_email, mock_datetime,
+                                       mock_api_logger, mock_mv_file, mock_load_file)
+
+    @mock.patch('seqr.views.apis.anvil_workspace_api.ANVIL_LOADING_DELAY_EMAIL', 'We are unable to load your data at this time.')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.ANVIL_LOADING_EMAIL_DATE', '2021-06-01')
+    @responses.activate
+    def test_add_workspace_data_loading_delay_email(
+            self, mock_send_email, mock_datetime, mock_api_logger, mock_tempfile, mock_mv_file, mock_slack,
+            mock_load_file, mock_airtable_logger, mock_utils_logger):
+
+        url = reverse(add_workspace_data, args=[PROJECT1_GUID])
+        url2 = reverse(add_workspace_data, args=[PROJECT2_GUID])
+
+        self._test_loading_delay_email(url, url2, REQUEST_BODY_ADD_DATA, REQUEST_BODY_ADD_DATA2, mock_send_email,
+                                       mock_datetime, mock_api_logger, mock_mv_file, mock_load_file)
+
+    def _test_loading_delay_email(self, url, url2, request_body, request_body2, mock_send_email, mock_datetime, mock_api_logger,
+                                  mock_mv_file, mock_load_file):
+        mock_load_file.return_value = LOAD_SAMPLE_DATA
+        mock_mv_file.return_value = True
         self.check_manager_login(url, login_redirect_url='/login/google-oauth2')
 
         # Test not yet anvil email date
         mock_send_email.side_effect = ValueError('Unable to send email')
         mock_datetime.strptime.side_effect = datetime.strptime
         mock_datetime.now.side_effect = lambda: datetime(2021, 3, 1, 0, 0, 0)
 
-        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY))
+        response = self.client.post(url, content_type='application/json', data=json.dumps(request_body))
         self.assertEqual(response.status_code, 200)
         mock_send_email.assert_not_called()
 
-        # Remove created project to allow future requests
-        project = Project.objects.get(
-            workspace_namespace=TEST_WORKSPACE_NAMESPACE, workspace_name=TEST_NO_PROJECT_WORKSPACE_NAME)
-        project.workspace_name = None
-        project.save()
-
         # Test after anvil email date
         mock_datetime.now.side_effect = lambda: datetime(2021, 9, 1, 0, 0, 0)
-        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY))
+        response = self.client.post(url2, content_type='application/json', data=json.dumps(request_body2))
```

Comment:
why does this need a different url now? Before this test worked if you called it with the same url

Reply:
Because we use two URLs to use two projects for the `not yet anvil email date` and `after anvil email date` tests, it can avoid some cleaning up of the project before starting the second test.

Reply:
I'd really rather not change test behavior unless actual behavior changed for this PR, so we can ensure that the functionality dud not actually change

Reply:
I can create two separate sharing tests for the `before` and `after` email dates. And then keep the test behavior unchanged for the `create__project_from_workspace` API.